### DISCUSSION TITLE
How to debug this CopyQ 'join' command?

### DISCUSSION THREAD

**Discussion by DeanintheMachine** [hluk/CopyQ#2790] [Q&A]
Happy CopyQ user here, getting my feet wet with commands. I pasted this example join command into my CopyQ version 8.0.0 on linux: https://github.com/hluk/copyq-commands/blob/master/Application/join-selected-items.ini

And here is the commands's code: 
copyq:
const separator = '\n';
var sel = new ItemSelection().current();
const texts = sel.itemsFormat(mimeText);
sel.selectAll();
add(texts.join(separator));
sel.invert();
selectItems(sel.rows()[0]);

The command does run, but instead of showing the joined entries, it shows this: 
ScriptableByteArray(0x55a84559d2f0)
ScriptableByteArray(0x55a84559c600)

I am guessing the culprit is the itemsFormat(mimeText) call? I dug a bit into the code, but the data structures are way over my head. 

Would anyone have an idea what may be going wrong?  Thanks!

**Comment by hluk** [comment]
I think it was fixed in version 9.0.0. Can you verify?

**Comment by sva-it** [comment]
The same problem on Arch Linux (CopyQ version 9.1.0):

ScriptableByteArray(0x5787a8c5fd80)
ScriptableByteArray(0x5787a8c5b2a0)

after join command

**Reply by hluk** [reply]
Oh, I can reproduce this on Qt 5 build of the app. The Qt 6 version does not have the problem. :thinking:

**Reply by hluk** [reply]
It is probably not worth fixing for the old Qt version. The problem is that you need to convert the items to string explicitly. I think `Array.prototype.join()` should do that; it is probably a bug in Qt 5.

You can convert the items to strings by calling `.map(str)` before the `.join(...)`:

```
add(texts.map(str).join(separator));
```

**Reply by sva-it** [reply]
Thank you very much. Your solution works for me. 👍
